### ISSUE TITLE
Implement and test GNN

### ISSUE THREAD

**Issue by kamil271e** [iwamaciek/stwp#3]
Start implementing solution that will further be expanded to look simirarly to GrapCast.

- [x] Transform the data into a graph format
- [x] Try [A3T-GCN](https://arxiv.org/abs/2006.11583)
- [x] Experiment with edge_weights
- [x] Experiment with edge_attr 
- [x] How to add vector features e.g. temporal encodings to a model ?
- [x] Encode temporal features (temporal encodings / embeddings)
- [x] Encode spatial features (positional encodings / embeddings)
- [x] Add other constant features - binary-land-sea mask, geopotential
- [x] ~~Edge embeddings~~
- [x] Establish a mesh-grid-like structure with enhanced connections among neighbors. A promising and straightforward starting point could involve adapting the solution implemented for creating neighbors in other baseline models
- [x] Compare linear layers and DenseGCNConv:  $MLP^{embedder}, MLP^{decoder}$
- [x] Layer Norm between graph cells
- [x] [Optional] ~~Try/read about heterogenous graphs: [source1](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.to_hetero_with_bases_transformer.to_hetero_with_bases) [source2](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.HeteroConv.html#torch_geometric.nn.conv.HeteroConv) [source3](https://www.youtube.com/watch?v=qL09oshDKww)~~
- [x] Test different GNN cells provided underneath
- [x] ~~Test different lr_scheduler - Cossine Annealing might be robust~~ feel free to try different ones but StepLR works fine

During the initial testing of new models, prioritize efforts to overfit on a single batch.

List of graph cells from torch geometric that supports edge_attr:
- [RGATConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.RGATConv.html#torch_geometric.nn.conv.RGATConv) 
- [GMMConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.GMMConv.html#torch_geometric.nn.conv.GMMConv)
- [SplineConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.SplineConv.html#torch_geometric.nn.conv.SplineConv)
- [NNConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.NNConv.html)
- [CGConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.CGConv.html#torch-geometric-nn-conv-cgconv) -  **I** 
- [ResGatedGraphConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.ResGatedGraphConv.html#torch_geometric.nn.conv.ResGatedGraphConv)
- [GATConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.GATConv.html#torch_geometric.nn.conv.GATConv) -  **I** 
- [GATv2Conv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.GATv2Conv.html#torch_geometric.nn.conv.GATv2Conv)
- [TransformerConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.TransformerConv.html#torch_geometric.nn.conv.TransformerConv) -  **I** 
- [GINEConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.GINEConv.html#torch_geometric.nn.conv.GINEConv)
- [GeneralConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.GeneralConv.html#torch_geometric.nn.conv.GeneralConv)
- [HEATConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.HEATConv.html#torch_geometric.nn.conv.HEATConv)
- [PNAConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.PNAConv.html#torch_geometric.nn.conv.PNAConv)
- [GENConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.GENConv.html#torch_geometric.nn.conv.GENConv) -  **I** 
- [PDNConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.PDNConv.html#torch_geometric.nn.conv.PDNConv) -  **I** 

 **I** - Implemented
